src/main/java/net/glowstone/entity/passive/GlowFishingHook.java

```diff
@@ -0,0 +1,243 @@
+
+        spawnMessage.add(new SpawnObjectMessage(getEntityId(), getUniqueId(),
+                SpawnObjectMessage.FISHING_HOOK, x, y, z, intPitch, intHeadYaw, 0, velocity));
+        return spawnMessage;
+    }
+
+    @Override
+    public void collide(Block block) {
+        // TODO
+    }
+
+    @Override
+    public void collide(LivingEntity entity) {
+        // No effect.
+    }
+
+    @Override
+    protected int getObjectId() {
+        return SpawnObjectMessage.FISHING_HOOK;
+    }
+
+    @Override
+    public boolean shouldSave() {
+        return false;
+    }
+
+    @Deprecated
+    @Override
+    public double getBiteChance() {
+        // Not supported in newer mc versions anymore
+        return 0;
+    }
+
+    @Deprecated
+    @Override
+    public void setBiteChance(double v) throws IllegalArgumentException {
+        // Not supported in newer mc versions anymore
+    }
+
+    private Entity getHookedEntity() {
+        return world.getEntityManager().getEntity(
+                metadata.getInt(MetadataIndex.FISHING_HOOK_HOOKED_ENTITY) - 1);
+    }
+
+    private void setHookedEntity(Entity entity) {
+        metadata.set(MetadataIndex.FISHING_HOOK_HOOKED_ENTITY,
+                entity == null ? 0 : entity.getEntityId() + 1);
+    }
+
+    @Override
+    public void pulse() {
+        super.pulse();
+        // TODO: Particles
+        // TODO: Bopper movement
+        if (location.getBlock().getType() == Material.WATER) {
+            increaseTimeLived();
+
+        }
+    }
+
+    private void increaseTimeLived() {
+        // "The window for reeling in when a fish bites is about half a second.
+        // If a bite is missed, the line can be left in the water to wait for another bite."
+        if (lived - lifeTime > 10) {
+            lifeTime = calculateLifeTime();
+            lived = 0;
+        }
+
+        // "If the bobber is not directly exposed to sun or moonlight,[note 1] the wait time will
+        // be approximately doubled.[note 2]"
+        Block highestBlockAt = world.getHighestBlockAt(location);
+        if (location.getY() < highestBlockAt.getLocation().getY()) {
+            if (ThreadLocalRandom.current().nextDouble(100) < 50) {
+                return;
+            }
+        }
+
+        if (GlowBiomeClimate.isRainy(location.getBlock())) {
+            if (ThreadLocalRandom.current().nextDouble(100) < 20) {
+                lived++;
+            }
+        }
+
+        lived++;
+    }
+
+    /**
+     * Removes this fishing hook. Drops loot and xp if a player is fishing.
+     */
+    public void reelIn() {
+        if (location.getBlock().getType() == Material.WATER) {
+            if (getShooter() instanceof Player) {
+                // TODO: Item should "fly" towards player
+                world.dropItemNaturally(((Player) getShooter()).getLocation(), getRewardItem());
+                ((Player) getShooter()).giveExp(ThreadLocalRandom.current().nextInt(1, 7));
```

Comment:
I think the `PlayerFishEvent` should be called and handled here, as part of this PR.

Reply:
Done, but I'm not 100% familiar with the events.